src/compas_robots/model/geometry.py

```diff
@@ -308,64 +260,39 @@ def data(self):
             "texture": self.texture.data if self.texture else None,
         }
 
-    @data.setter
-    def data(self, data):
-        self.name = data["name"]
-        self.color = Color.from_data(data["color"]) if data["color"] else None
-        self.texture = Texture.from_data(data["texture"]) if data["texture"] else None
+    @classmethod
+    def from_data(cls, data):
+        return cls(
+            name=data["name"],
+            color=Color.from_data(data["color"]) if data["color"] else None,
+            texture=Texture.from_data(data["texture"]) if data["texture"] else None,
+        )
 
     def get_color(self):
         """Get the RGBA color array of the material.
 
         Returns
         -------
-        [float, float, float, float]
-            List of 4 floats (``0.0-1.0``) indicating RGB colors and Alpha channel of the material.
+        :class:`~compas.colors.Color`
+            Color of the material.
 
         Examples
         --------
         >>> material = Material('aqua')
         >>> material.get_color()
-        (0.0, 1.0, 1.0, 1.0)
+        Color(0.0, 1.0, 1.0, alpha=1.0)
 
         """
         if self.name:
             try:
-                color = compas.colors.Color.from_name(self.name)
-                return color.rgba
+                return Color.from_name(self.name)
             except ValueError:
                 pass
-        if self.color:
-            return self.color.rgba
-        return None
-
-
-TYPE_CLASS_ENUM = {
-    "box": compas.geometry.Box,
-    "cylinder": compas.geometry.Cylinder,
-    "sphere": compas.geometry.Sphere,
-    "capsule": compas.geometry.Capsule,
-    "mesh": MeshDescriptor,
-}
-
-# TYPE_CLASS_ENUM_BY_DATA = {
-#     ("frame", "xsize", "ysize", "zsize"): compas.geometry.Box,
-#     ("circle", "height"): compas.geometry.Cylinder,
-#     ("point", "radius"): compas.geometry.Sphere,
-#     ("line", "radius"): compas.geometry.Capsule,
-#     ("attr", "filename", "scale"): MeshDescriptor,
-#     ("attr", "filename", "meshes", "scale"): MeshDescriptor,
-# }
-
 
-# def _get_type_from_shape_data(data):
-#     # This is here only to support models serialized with older versions of COMPAS
-#     if "type" in data:
-#         return TYPE_CLASS_ENUM[data["type"]]
+        if self.color:
+            return Color(*self.color.rgba)
 
-#     # The current scenario is that we need to figure out the object type based on the DATASCHEMA
-#     keys = tuple(sorted(data.keys()))
-#     return TYPE_CLASS_ENUM_BY_DATA[keys]
+        return None
 
 
 class Geometry(Data):
```

Comment:
I know this isn't new but given that there's a `Geometry` in core maybe we should at some point rename this to `LinkGeometry`?

Reply:
Good idea! Renamed at https://github.com/compas-dev/compas_robots/pull/1/commits/177312b75e81e2018d1dcde234f4778cb38bc402